lms/djangoapps/support/views/program_enrollments.py

```diff
@@ -262,3 +262,36 @@ def _get_enrollments(self, user=None, external_user_key=None):
         ).prefetch_related('program_course_enrollments')
         serialized = ProgramEnrollmentSerializer(program_enrollments, many=True)
         return serialized.data
+
+
+class SAMLProvidersWithOrg(APIView):
+    """
+    Support-only API View for fetching a list of all organization keys.
+    """
+    @method_decorator(require_support_permission)
+    def get(self, request):
+        """
+        The get request returns a list of all organization keys.
+        * Example Request:
+            - GET /support/get_saml_providers_with_org_key/
+        * Example Response:
+            [
+                'test_org',
+                'donut_org',
+                'tri_org'
+            ]
+        """
+        saml_providers_with_org_key = self._get_org_keys_and_idps()
+        return Response(data=saml_providers_with_org_key)
+
+    def _get_org_keys_and_idps(self):
+        """
+        From our Third_party_auth models, return a list of
+        of organizations keys.
+        """
+        saml_providers = SAMLProviderConfig.objects.current_set().filter(
+            enabled=True,
+            organization__isnull=False
+        ).select_related('organization')
+
+        return [saml_provider.organization.short_name for saml_provider in saml_providers]
```

Comment:
There is a sort of to-and-fro between org key and short name in code and naming conventions. Let's use a single term across.